crates/rolldown/src/module_loader/normal_module_task.rs

```diff
@@ -158,7 +158,7 @@ impl NormalModuleTask {
 
     let side_effects = match hook_side_effects {
       Some(side_effects) => match side_effects {
-        HookSideEffects::True => DeterminedSideEffects::UserDefined(true),
+        HookSideEffects::True => lazy_check_side_effects(),
```

Comment:
> If false is returned for moduleSideEffects and no other module imports anything from this module, then this module will not be included even if the module would have side effects.

> If true is returned, Rollup will use its default algorithm to include all statements in the module that have side effects (such as modifying a global or exported variable).

Reply:
Will the plain import `import './foo.js'` be reserved if `foo.js` is `HookSideEffects::True` and it actually doesn't have side effect.